packages/labs/testing/src/lib/fixtures/csrFixture.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {render} from 'lit';
+import type {LitElement, TemplateResult} from 'lit';
+import type {FixtureOption} from './fixtureOption.js';
+
+/**
+ * Renders the provided lit-html template with a Lit element client-side.
+ *
+ * @param {TemplateResult} template - lit-html template. Must contain a single
+ * top level custom element.
+ * @param {string[]} option.modules - Path to custom element definition modules
+ * needed to render template, relative to the test file.
+ * @param {string} [option.base] - Optional. Base path for the modules.
+ * Generally should be `import.meta.url`. Will guess from stack trace if not
+ * provided.
+ */
+export async function csrFixture<T extends LitElement>(
+  template: TemplateResult,
+  {modules, base}: FixtureOption
+): Promise<T> {
+  if (base === undefined) {
+    // Find the test file url from the call stack
```

Comment:
This is quite clever! But maybe kind of surprising? I wouldn't realy expect a function to behave differently depending on where it was called from. Is there some precedent for this, like something `@open-wc/testing` also does?

Another idea could be to just always require the base url be manually specified.

Reply:
I had originally made the base required but I added this purely as a convenience for users. This should do the same thing as the user providing `import.meta.url` as the base which I felt like was going to be the vast majority of use cases.

I have not seen any precedent for this and this need specifically arose for this use case because we need be able to resolve filepaths provided as a string to dynamically import from a different module/environment. Our `renderModule` from ssr requires `import.meta.url` to be passed in for the same reason.